firewood/src/merkle.rs

```diff
@@ -66,6 +66,10 @@ macro_rules! write_node {
     };
 }
 
+pub trait NodeSource {
+    fn get_node(&self, ptr: DiskAddress) -> Result<NodeObjRef, MerkleError>;
+}
```

Comment:
This seems a little odd to me (probably because there's only a single implementation right now). What, other than a `Merkle<S, T>` would ever implement this trait? 

Seems to me that a `DbRev` and a `Proposal` would always create a stream by talking to their underlying `Merkle<S, T>`?

I think it's a lot easier to find the right abstraction when you have at least one more implementation.

And I'm not saying this is wrong, I can only go on my intuition without seeing another use.

Reply:
The issue is that iterators need to have a reference back to something that holds nodes, since `get_node` returns a reference to a node with a lifetime of their parent. Iterators keep around either a `DbRev` or a `Proposal`, not a`merkle` per-se. In order for the iterator to live long enough between a RPC call, a reference to something that lives longer than that is needed.

I am open for other ideas on how to fix this, maybe we can talk about this during standup.

Reply:
Yeah, happy to talk on a call. By no means do I think my comment should hold up the PR, just some food for thought. I don't see why a `Proposal` couldn't return a `MerkleKeyValueStream` that's not generic over a `Merkle<S, T>`, though. Maybe that means that the underlying `DbRev` has to expose that, but the actual `get_node` functionality actually always comes from the `Store`. From what I understand, the store will always be wrapped in some `Merkle<_, T>`, unless there's something else that's generic over the store that I'm just missing (very possible).